lib/perl/Genome/InstrumentData/AlignmentResult/Merged.pm

```diff
@@ -219,24 +226,70 @@ sub create {
         for my $alignment (@alignments) {
             $alignment->add_user(user => $self, label => 'uses');
         }
-
-        return 1;
-    };
-    if (my $error = $@) {
-        $tx->rollback();
-        die $error;
-    } elsif ($rv ne 1) {
+    }
+    catch {
         $tx->rollback();
-        $self->error_message('Unexpected return value: ' . $rv);
-        die $self->error_message;
-    } else {
-        $tx->commit();
+        die $self->error_message('Merge failed due to error: ' . $_);
     }
+    finally {
+        for my $allocation (@temp_allocations) {
+            $allocation->delete;
+        }
+        unless ($_) {
+            $tx->commit();
```

Comment:
Having multiple finally blocks and depending on their execution order seems like a code smell.  The docs for Try::Tiny don't say anything about the order, and includes this
```
You must always do your own error handling in the finally block. "Try::Tiny" will not do anything about handling possible errors coming from code located in these blocks.
```
Which sounds like you can't even count on all the finally blocks running if you throw an exception in one of them.

Using nested try/catch/finally blocks would be safer and might be more clear about what you're trying to do and when.